src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -288,7 +288,7 @@ bool CanTokenFollowTypeInPattern(Precedence precedence)
                     SyntaxKind.CloseParenToken,
                     static @this => @this.IsPossibleSubpatternElement(),
                     static @this => @this.ParseSubpatternElement(),
-                    SkipBadPatternListTokens,
+                    _termState.HasFlag(TerminatorState.IsExpressionOrPatternInCaseLabelOfSwitchStatement) ? SkipBadPatternListTokens_ExitOnColon : SkipBadPatternListTokens_Normal,
```

Comment:
this seems unnecessary (to me).   Just pass SkipBadPatternListTokens in, and have it dispatch based on the _termState.